paddle/fluid/distributed/collective/ProcessGroup.cc

```diff
@@ -44,5 +51,13 @@ ProcessGroup::ProcessGroup(int rank, int size, const platform::Place& place,
   }
 }
 
+ProcessGroup::ProcessGroup(int rank, int size, int gid)
+    : rank_(rank), size_(size), gid_(gid) {
+  if (gid != IGNORE_ID) {
+    auto map = ProcessGroupMapFromGid::getInstance();
+    map->insert(gid_, this);
```

Comment:
The construct function can't return value, so it may be wrapped by `Init` function or use `PADDLE_ENFORCE_XXX`